crates/oxc_linter/src/rules/eslint/func_names.rs

```diff
@@ -0,0 +1,507 @@
+use oxc_ast::ast::{
+    AssignmentTarget, AssignmentTargetProperty, BindingPatternKind, Expression, Function,
+    FunctionType, PropertyKind,
+};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct FuncNames {
+    default_config: FuncNamesConfig,
+    generators_config: FuncNamesConfig,
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+enum FuncNamesConfig {
+    #[default]
+    Always,
+    AsNeeded,
+    Never,
+}
+
+impl FuncNamesConfig {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "always" => FuncNamesConfig::Always,
+            "as-needed" => FuncNamesConfig::AsNeeded,
+            "never" => FuncNamesConfig::Never,
+            _ => FuncNamesConfig::default(),
+        }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Require or disallow named function expressions
+    ///
+    /// ### Why is this bad?
+    ///
+    /// If you leave off the function name then when the function throws an exception you are likely
+    /// to get something similar to anonymous function in the stack trace. If you provide the optional
+    /// name for a function expression then you will get the name of the function expression in the stack trace.
+    ///
+    /// /// ### Example
+    ///
+    /// ```javascript
+    /// Foo.prototype.bar = function bar() {};
+    /// ```
+    FuncNames,
+    style,
+    pending
+);
+/**
+ * Determines whether the current FunctionExpression node is a get, set, or
+ * shorthand method in an object literal or a class.
+ */
+fn is_object_or_class_method(parent_node: Option<&AstNode>) -> bool {
+    if parent_node.is_none() {
+        return false;
+    }
+
+    match parent_node.unwrap().kind() {
+        AstKind::MethodDefinition(_) => true,
+        AstKind::ObjectProperty(property) => {
+            property.method
+                || property.kind == PropertyKind::Get
+                || property.kind == PropertyKind::Set
+        }
+
+        _ => false,
+    }
+}
+/**
+ * Determines whether the current FunctionExpression node has a name that would be
+ * inferred from context in a conforming ES6 environment.
+ */
+fn has_inferred_name(function: &Function, parent_node: Option<&AstNode>) -> bool {
+    if is_object_or_class_method(parent_node) {
+        return true;
+    }
+
+    if parent_node.is_none() {
```

Comment:
Redundant, since the above function already performs this check.

Reply:
If you need it to be some, consider doing 

`let Some(foo) = bar else { return; };` above.